python/pyspark/mllib/tests.py

```diff
@@ -1041,10 +1054,12 @@ def test_trainOn_model(self):
         self.ssc.start()
 
         # Give enough time to train the model.
-        self._ssc_wait(t, 6.0, 0.01)
-        finalModel = stkm.latestModel()
-        self.assertTrue(all(finalModel.centers == array(initCenters)))
-        self.assertEquals(finalModel.clusterWeights, [5.0, 5.0, 5.0, 5.0])
+        def termCheck():
+            finalModel = stkm.latestModel()
+            all(finalModel.centers == array(initCenters)) and \
+            finalModel.clusterWeight == [5.0, 5.0, 5.0, 5.0]
+        self._ssc_wait_checked(t, 20.0, termCheck)
+        self.assertTrue(termCheck())
```

Comment:
There is still a slight possibility that between the last time term_check() is called in the _ssc_wait_checked, and next time its called in this method, another batch may have been processed, which which fail the test unnecessarily. So a better approach would be for the _ssc_wait_checked method to return True if the term_check() has succeeded within the timeout, otherwise return false. Then there is not need to check term_check() once again.